code/game/objects/items/RCD.dm

```diff
@@ -1133,53 +1164,96 @@ GLOBAL_VAR_INIT(icon_holographic_window, init_holographic_window())
 	return ..()
 
 /obj/item/construction/plumbing/attack_self(mob/user)
-	..()
-	if(!choices.len)
-		for(var/obj/machinery/plumbing/plumbing_type as anything in plumbing_design_types)
-			choices += list(initial(plumbing_type.name) = image(initial(plumbing_type.icon), icon_state = initial(plumbing_type.icon_state)))
-			name_to_type[initial(plumbing_type.name)] = plumbing_type
-			machinery_data["cost"][plumbing_type] = plumbing_design_types[plumbing_type]
+	ui_interact(user)
 
-	// Update duct icon
-	var/image/duct_image = choices["fluid duct"]
-	duct_image.color = current_color
+/obj/item/construction/plumbing/examine(mob/user)
+	. = ..()
+	. += "You can scroll your mouse wheel to change the piping layer."
+	. += "You can right click a fluid duct to set the Plumbing RPD to its color and layer."
+
+/obj/item/construction/plumbing/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "PlumbingService", name)
+		ui.open()
+
+/obj/item/construction/plumbing/ui_assets(mob/user)
+	return list(
+		get_asset_datum(/datum/asset/spritesheet/plumbing),
+	)
 
-	var/choice = show_radial_menu(user, src, choices, custom_check = CALLBACK(src, PROC_REF(check_menu), user), require_near = TRUE, tooltips = TRUE)
-	if(!choice || !check_menu(user))
+/obj/item/construction/plumbing/ui_static_data(mob/user)
+	var/list/data = list("paint_colors" = GLOB.pipe_paint_colors)
+	return data
```

Comment:
```suggestion
	return list("paint_colors" = GLOB.pipe_paint_colors)
```

Reply:
Copied this code from RPD . I'll change it there as well then